lighthouse-core/lib/traces/tracing-processor.js

```diff
@@ -177,57 +121,60 @@ class TraceProcessor {
   /**
    * Calculates the maximum queueing time (in ms) of high priority tasks for
    * selected percentiles within a window of the main thread.
-   * @see https://docs.google.com/document/d/1b9slyaB9yho91YTOkAQfpCdULFkZM9LqsipcX3t7He8/preview
-   * @param {!traceviewer.Model} model
-   * @param {{traceEvents: !Array<!Object>}} trace
+   * @see https://docs.google.com/document/d/18gvP-CBA2BiBpi3Rz1I1ISciKGhniTSZ9TY0XCnXS7E/preview
+   * @param {!TraceOfTabArtifact} tabTrace
    * @param {number=} startTime Optional start time (in ms) of range of interest. Defaults to trace start.
    * @param {number=} endTime Optional end time (in ms) of range of interest. Defaults to trace end.
    * @param {!Array<number>=} percentiles Optional array of percentiles to compute. Defaults to [0.5, 0.75, 0.9, 0.99, 1].
    * @return {!Array<{percentile: number, time: number}>}
    */
-  static getRiskToResponsiveness(model, trace, startTime, endTime, percentiles) {
+  static getRiskToResponsiveness(tabTrace, startTime, endTime, percentiles) {
+    const navStart = tabTrace.navigationStartEvt.ts;
+    const traceStart = tabTrace.earliestTraceEvt.ts;
+    const traceEnd = tabTrace.latestTraceEvt.ts;
+
     // Range of responsiveness we care about. Default to bounds of model.
-    startTime = startTime === undefined ? model.bounds.min : startTime;
-    endTime = endTime === undefined ? model.bounds.max : endTime;
+    startTime = startTime === undefined ? (traceStart - navStart) / 1000 : startTime;
```

Comment:
> better question, should we just default these bounds to navstart and traceEnd now instead? negative start time doesn't really make a whole lot of sense and we take it to be the beginning of time everywhere we consume this data

ha, sorry I was too slow to respond :)

I think it would be better to still have `getMainThreadTopLevelEvents` and `getMainThreadTopLevelEventDurations` return all events, even ones before navStart.

Our current uses don't need anything before then, but we'll hopefully be doing more sophisticated trace analysis in the future (e.g. over multiple navigations, though we'll have several other complications to figure out before then :) and I think having them return all of the main thread/top level events in the trace will be what you would expect from them.

Having a note that the times are relative to navStart (so negative times are just before that) doesn't seem too confusing.